### PR TITLE
Fixes #21973 - maintenance mode off fails if no custom rule exists

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

**Comment by kgaikwad** [comment]
@swapab , PR updated with new rubocop changes.

**Comment by kgaikwad** [comment]
@iNecas & @swapab ,

If the changes are good to go, should I merge this pull request?

File: definitions/features/foreman_1_7_x.rb [hunk 1]
```diff
@@ -35,7 +35,10 @@ def custom_iptables_chain(name, rules)
 
   def del_custom_iptables_chain(name)
     return unless execute?("iptables -L #{name}") # the chain is already gone
-    execute!("iptables -D INPUT -j #{name}")
+    output =  execute("iptables -L INPUT | awk '$1 ~ /#{name}/ { print }'").strip
+    unless output.empty?
```
Comment (iNecas) [inline definitions/features/foreman_1_7_x.rb]: what about using this as a condition: `execute?("iptables -L INPUT | tail -n +3 | grep '^#{name} '")`

File: definitions/features/foreman_1_7_x.rb [hunk 2]
```diff
@@ -35,7 +35,10 @@ def custom_iptables_chain(name, rules)
...
+      execute("iptables -D INPUT -j #{name}")
```
Comment (iNecas) [inline definitions/features/foreman_1_7_x.rb]: I would still want to keep `execute!` here, to know if there is some other issue with this in the future.

**Comment by kgaikwad** [comment]
@iNecas,
Updated PR with suggested changes.
Ready for review

**Comment by iNecas** [comment]
Thanks @kgaikwad

**Pull request merged**